pkg/partition/partition.go

```diff
@@ -164,3 +165,44 @@ func (d *Disk) CreateSinglePartition() error {
 	klog.Infof("created a single partition on disk %s", d.DevPath)
 	return nil
 }
+
+// CreatePartitionTable create a GPT header on the disk
+func (d *Disk) CreatePartitionTable() error {
+	if !features.FeatureGates.IsEnabled(features.GPTBasedUUID) {
```

Comment:
Is this check required here?  This is a utility package. The functionality of this package should not depend on the feature flags that are used. The caller of this function should be responsible for making sure the flag is enabled and then call this function.